File: Classes/Structure/Store/level.vb
''' <summary>
''' DataType representing any number of Shelf's as a Level().
''' </summary>
Public Class level
	Public Shared arrayShelf() As Shelf
	Private stringPathLevel As String
	Private integerShelfCounter As Integer
''' <summary>
''' Empty Sub.
''' </summary>
	Public Sub New()
		
	End Sub
''' <summary>
''' Get/Adjust the path of the Level without Initialize() and Load().
''' </summary>
	Public Property LevelPath As String
		Get
			Return stringPathLevel
		End Get
		Set (Value As String)
			stringPathLevel = Value
		End Set
	End Property
''' <summary>
''' Resizes arrayShelf() and fill with New Shelf's.
''' </summary>
	Public Sub ShelfInitialize()
		integerShelfCounter = 0
		ReDim arrayShelf(System.IO.Directory.GetDirectories(stringPathLevel, "*", System.IO.SearchOption.TopDirectoryOnly).GetUpperBound(0))
		For Each folder As String In System.IO.Directory.GetDirectories(stringPathLevel, "*", System.IO.SearchOption.TopDirectoryOnly)
			arrayShelf(integerShelfCounter) = New Shelf
			arrayShelf(integerShelfCounter).ShelfPath = folder & "\"
			integerShelfCounter += 1
		Next
	End Sub
''' <summary>
''' Loads all Shelves with all Bin's with all Items.
''' </summary>
	Public Shared Sub ShelfLoad()
		For Each item As Shelf In arrayShelf
			item.ShelfInitialize()
			item.ShelfLoad()
		Next
	End Sub
''' <summary>
''' Adds a empty Level with 1 Shelf and 1 Bin containing 'Nothing'.
''' </summary>
	Public Sub ShelfAdd()
		Dim intCheck As Integer = 0
		Do
			If System.IO.Directory.Exists(stringPathLevel & "\" & intCheck) = False Then
				Exit Do
			End If
			intCheck += 1
		Loop
		ReDim Preserve arrayShelf(intCheck)
		System.IO.Directory.CreateDirectory(stringPathLevel & "\" & intCheck)
		arrayShelf(arrayShelf.GetUpperBound(0)) = New Shelf
		arrayShelf(arrayShelf.GetUpperBound(0)).ShelfPath = stringPathLevel & "\" & intCheck
	End Sub
End Class

File: Game/Classes/Structure/Store/level.cs
using System;

/// <summary>
/// DataType representing any number of Shelf's as a Level().
/// </summary>
public class level
{
    public static shelf[] arrayShelf;
    private string stringPathLevel;
    private int integerShelfCounter;

    /// <summary>
    /// Empty Sub.
    /// </summary>
    public level()
    {

    }

    /// <summary>
    /// Get/Adjust the path of the Level without Initialize() and Load().
    /// </summary>
    public string LevelPath {
        get { return stringPathLevel; }
        set { stringPathLevel = value; }
    }

    /// <summary>
    /// Resizes arrayShelf() and fill with New Shelf's.
    /// </summary>
    public void ShelfInitialize()
    {
        integerShelfCounter = 0;
        arrayShelf = new shelf[System.IO.Directory.GetDirectories(stringPathLevel, "*", System.IO.SearchOption.TopDirectoryOnly).GetUpperBound(0) + 1];
        foreach (string folder in System.IO.Directory.GetDirectories(stringPathLevel, "*", System.IO.SearchOption.TopDirectoryOnly)) {
            arrayShelf[integerShelfCounter] = new shelf();
            arrayShelf[integerShelfCounter].ShelfPath = folder + "\\";
            integerShelfCounter += 1;
        }
    }
    
    /// <summary>
    /// Loads all Shelves with all Bin's with all Items.
    /// </summary>
    public static void ShelfLoad()
    {
        foreach (shelf item in arrayShelf) {
            item.ShelfInitialize();
            item.ShelfLoad();
        }
    }
    
    /// <summary>
    /// Adds a empty Level with 1 Shelf and 1 Bin containing 'Nothing'.
    /// </summary>
    public void ShelfAdd()
    {
        int intCheck = 0;
        do {
            if (System.IO.Directory.Exists(stringPathLevel + "\\" + intCheck)) {
            } else {
                break; // TODO: might not be correct. Was : Exit Do
            }
            intCheck += 1;
        } while (true);
        Array.Resize(ref arrayShelf, intCheck + 1);
        System.IO.Directory.CreateDirectory(stringPathLevel + "\\" + intCheck);
        arrayShelf[arrayShelf.GetUpperBound(0)] = new shelf();
        arrayShelf[arrayShelf.GetUpperBound(0)].ShelfPath = stringPathLevel + "\\" + intCheck;
    }
}